### PR TITLE
chore: oauth v2 stats refactor

### PR DESCRIPTION
**Pull request by sanpj2292**
# Description

The way in which stats are written is not easily readable. Through this PR, we are trying to make the stat emission in oauthv2 module a bit more developer friendly and readable

## Linear Ticket

Resolves INT-2744

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: services/oauth/v2/oauth.go [hunk 1]
```diff
@@ -342,14 +348,15 @@ func (h *OAuthHandler) GetRefreshTokenErrResp(response string, accountSecret *Ac
 // This method hits the Control Plane to get the account information
 // As well update the account information into the destAuthInfoMap(which acts as an in-memory cache)
 func (h *OAuthHandler) fetchAccountInfoFromCp(refTokenParams *RefreshTokenParams, refTokenBody RefreshTokenBodyParams,
-	authStats *OAuthStats, logTypeName string,
+	statsHandler OAuthStatsHandler, logTypeName string,
 ) (int, *AuthResponse, error) {
+	actionType := strings.Join(strings.Fields(strings.ToLower(logTypeName)), "_")
```
Comment (sanpj2292) [inline services/oauth/v2/oauth.go:353]:
```Go
> strings.Join(strings.Fields(strings.ToLower("Refresh Token")), "_")
refresh_token
```

File: services/oauth/v2/stats.go [hunk 2]
```diff
@@ -18,40 +19,57 @@ type OAuthStats struct {
...
+}
+
+func (m *OAuthStatsHandler) mergeTags(tags stats.Tags) stats.Tags {
+	allTags := m.defaultTags
+	for key, value := range tags {
+		allTags[key] = value
 	}
-	s.stats.NewTaggedStat(s.statName, stats.CountType, statsTags).Increment()
+	return allTags
+}
```
Comment (Sidddddarth) [inline services/oauth/v2/stats.go:59]: This should be the other way around. If we don't want to put values into `m.defaultTags`.
```
func (m *OAuthStatsHandler) mergeTags(tags stats.Tags) stats.Tags {
	for key, value := range m.defaultTags {
		tags[key] = value
	}
	return tags
}
```
Add a map lookup check if you don't want to override with default.

File: services/oauth/v2/stats.go [hunk 3]
```diff
@@ -18,40 +19,57 @@ type OAuthStats struct {
...
+
+func (m *OAuthStatsHandler) getStatName(suffix string) string {
+	return strings.Join([]string{"oauth_action", suffix}, "_")
+}
```
Comment (Sidddddarth) [inline services/oauth/v2/stats.go:63]: minor
Can inline this function's call.

File: services/oauth/v2/stats.go (hunk 2 above)
Comment (sanpj2292) [reply services/oauth/v2/stats.go]: Used a different way to solve this problem. Please check and let me know if it works

File: services/oauth/v2/stats.go (hunk 3 above)
Comment (sanpj2292) [reply services/oauth/v2/stats.go]: Done

**Review by Sidddddarth — APPROVED** [review]

**Review by koladilip — APPROVED** [review]

**Pull request merged**